scrapy/http/request/__init__.py

```diff
@@ -50,8 +50,8 @@ def _set_url(self, url):
         if not isinstance(url, six.string_types):
             raise TypeError('Request url must be str or unicode, got %s:' % type(url).__name__)
 
-        url = to_native_str(url, self.encoding)
-        self._url = escape_ajax(safe_url_string(url))
+        s = safe_url_string(url, self.encoding)
```

Comment:
@kmike , yes, if they want to use correct URL escaping and if it's not UTF8, then they need pass the information at some point. We should do the work for them with `FormRequest.from_response`, and making the URL safe in `Response.urljoin` (amending this PR or another).
The hard thing here is that correct encoding for query part may depend on form information within the document, document that may not have been parsed for this.
I believe for `CrawlSpider`, getting encoding right can be achieved with `encoding` information being kept with `Link` until `Request` is created from it.